File: frontend/src/domain/dtos/GenerateTextResponseDto.ts
// Keep in sync with backend/VocabuAI.Application/Learning/Generation/GenerateTextResponseDto.cs
import { Language } from "../Language";
import type { AiProvider } from "../AiProvider";

export type GenerateTextResponseDto = {
  language: Language;
  text: string;
  isValid: boolean;
  errorMessage: string | null;
  provider: AiProvider;
  tokenUsage: {
    promptTokens: number;
    completionTokens: number;
    totalTokens: number;
  } | null;
  usagePercent: number | null;
};

File: backend/VocabuAI.Application/Learning/Generation/GenerateTextResponseDto.cs
using VocabuAI.Application.Learning.Generation.Contracts;

namespace VocabuAI.Application.Learning.Generation;

// Keep in sync with frontend/src/domain/dtos/GenerateTextResponseDto.ts

public sealed record GenerateTextResponseDto(
    Language Language,
    string Text,
    bool IsValid,
    string? ErrorMessage,
    AiProvider Provider,
    AiTokenUsage? TokenUsage,
    double? UsagePercent
);
